kbs/src/api/src/attestation/intel_trust_authority/mod.rs

```diff
@@ -82,18 +88,23 @@ impl Attest for IntelTrustAuthority {
             .await
             .map_err(|e| anyhow!("Post attestation request failed: {:?}", e))?;
 
-        if resp.status() != reqwest::StatusCode::OK {
+        let status = resp.status();
+        if status != reqwest::StatusCode::OK {
+            let body = resp
+                .json::<ErrorResponse>()
+                .await
+                .map_err(|e| anyhow!("Deserialize error response failed: {:?}", e))?;
             bail!(
-                "Attestation request failed: respone status={}",
-                resp.status()
+                "Attestation request failed: response status={}, message={}",
+                status, body.error.unwrap_or_else(|| body.message.unwrap_or_else(|| "-".to_string()))
```

Comment:
I think you mean
```suggestion
                body.error.unwrap_or("-".to_string()), body.message.unwrap_or("-".to_string())
```
?

Reply:
For testing I'm using this instance: https://api-poc-user1.project-amber-smas.com
It should always respond with error property but initial call after a while responds with "message" rather than expected "error" -> they both are never accessible in the same time

Reply:
@Xynnn007, the main thing is that we'll have have **body.error** or **body.message**, but never **both**.
With that in mind, I think @pawelpros' code does the correct thing, trying to unwrap the first, and if that fails, goes for the second.

Reply:
Let's go with expected error format -> that "message" is probably specific for env and for now simpliest solution should do the trick.